nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp

```diff
@@ -423,20 +440,15 @@ VBufStorage_fieldNode_t* GeckoVBufBackend_t::fillVBuf(
 	previousNode=NULL;
 
 	//get IA2Attributes -- IAccessible2 attributes;
-	BSTR IA2Attributes;
 	map<wstring,wstring> IA2AttribsMap;
-	if(pacc->get_attributes(&IA2Attributes)==S_OK) {
-		IA2AttribsToMap(IA2Attributes,IA2AttribsMap);
-		SysFreeString(IA2Attributes);
-		// Add each IA2 attribute as an attrib.
-		for(map<wstring,wstring>::const_iterator it=IA2AttribsMap.begin();it!=IA2AttribsMap.end();++it) {
-			s<<L"IAccessible2::attribute_"<<it->first;
-			parentNode->addAttribute(s.str(),it->second);
-			s.str(L"");
-		}
-	} else
-		LOG_DEBUG(L"pacc->get_attributes failed");
 	map<wstring,wstring>::const_iterator IA2AttribsMapIt;
+	populateMapWithIA2AttributesFromPacc(pacc, IA2AttribsMap);

```

Comment:
Given that the return value is not used, I'd prefer this to internally construct the map and return it. If the error condition needs to be signaled then I'd suggest using `std::optional` to do so.